### PR TITLE
OS12814968: ASSERTION:(lib\runtime\Base/FunctionBody.h) IsFunctionBody()

### PR DESCRIPTION
**Pull request by suwc**
Function body not available for deferred-parse function.
Add additional checking.

### CODE REVIEW

File: lib/Runtime/Base/FunctionInfo.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ namespace Js
     FunctionBody *
     FunctionInfo::GetFunctionBody() const
     {
-        return functionBodyImpl == nullptr ? nullptr : functionBodyImpl->GetFunctionBody();
+        return (functionBodyImpl && functionBodyImpl->IsFunctionBody()) ? functionBodyImpl->GetFunctionBody() : nullptr;
```
Comment (agarwal-sandeep) [inline lib/Runtime/Base/FunctionInfo.cpp]: There is no harm in putting this additional check here as we already have an assert in FunctionProxy::GetFunctionBody but I think we should rather fix JsDiagGetFunctionPosition in either of 2 ways
1. If the function is not parsed, parse it and then return the information (preferable as user have no knowledge of defer-parse or re-deferal)
2. Check if function have body and if not return failure code

Comment (curtisman) [inline lib/Runtime/Base/FunctionInfo.cpp]: Agreed.  I'd much rather you do that in JsDiagGetFunctionPosition and separately change the Assert in GetFunctionBody to fail fast instead.  Or change the API to always to TryGetFunctionBody which a check and make the API explicit, and eliminate all the extra check in the caller.

File: lib/Jsrt/JsrtDiag.cpp [hunk 2]
```diff
@@ -498,6 +498,10 @@ CHAKRA_API JsDiagGetFunctionPosition(
         }
 
         Js::ScriptFunction* jsFunction = Js::ScriptFunction::FromVar(function);
+        if (!jsFunction->GetParseableFunctionInfo()->IsFunctionParsed())
+        {
+            Js::JavascriptFunction::DeferredParse(&jsFunction);
```
Comment (agarwal-sandeep) [inline lib/Jsrt/JsrtDiag.cpp]: Use DeferredParseCore and assert on fParsed

Comment (agarwal-sandeep) [inline lib/Jsrt/JsrtDiag.cpp]: @pleath will this work for all nested and re-deferral scenarios?

**Pull request merged**